lib/broccoli/ember-app.js

```diff
@@ -187,7 +187,8 @@ class EmberApp {
     let srcPath = this._resolveLocal('src');
     let srcTree = existsSync(srcPath) ? new WatchedDir(srcPath) : null;
 
-    let appTree = new WatchedDir(this._resolveLocal('app'));
+    let appPath = this._resolveLocal('app');
+    let appTree = existsSync(appPath) ? new WatchedDir(appPath) : null;
```

Comment:
Ya, we need to flag this, and add tests to cover these changes generally speaking.